packages/auth/amplify_auth_cognito/lib/src/auth_plugin_impl.dart

```diff
@@ -72,6 +72,15 @@ class AmplifyAuthCognito extends AuthPluginInterface implements Closeable {
       );
     }
 
+    // Configure this plugin to act as a native iOS/Android plugin.
+    if (!zIsWeb && (Platform.isAndroid || Platform.isIOS)) {
+      final nativePlugin = _NativeAmplifyAuthCognito(this);
+      NativeAuthPlugin.setup(nativePlugin);
+
+      final nativeBridge = NativeAuthBridge();
+      await nativeBridge.configure();
+    }
```

Comment:
Maybe it would be helpful to add some comments explaining why NativeAuthPlugin only provides FetchAuthSession functionality as the naming suggests it would have handled all Auth API related method calls.

Reply:
Is there a design document on our use of pigeon?

Reply:
We don't have a design doc for using pigeon. It's being used throughout, though, including in secure storage as well. What info can I provide? All it does is generate the boilerplate to talk to iOS/Android. The code would be the same if we were to write it by hand, so the design is technically the same.

I will add some notes for the interface (why only fetchAuthSession)